### ISSUE TITLE
stat.inode is not reliable as a check for samefile() if using network files on Windows 10  

### ISSUE THREAD

**Issue by sunito** [JuliaLang/julia#50596]
I was baffled for quite a while, when I found that  samefile("N:\\A", "N:\\B") can return **true** for (_different_) network files on Windows 10 running in VirtualBox.
I then found, that stat(file_name) does not yield consistent .inode values.

Steps to reproduce: 
1. Start Windows 10 in a VirtualBox with a (writable) shared folder. 
   Let's assume that this folder is mounted to drive "N:".  
   
2. Check, that they are not constant:
```
julia> [stat("N:\\").inode for i in 1:9]
  9-element Vector{UInt64}:
   0x00000000a6a14aa0
   0x00000000ac184aa0
   0x00000000a6973aa0
   0x00000000a1596010
   0x00000000a69caaa0
   0x00000000a276a840
   0x000000009dc7a5e0
   0x00000000a6974aa0
   0x00000000a69dbaa0
```
4. Check that some inode values repeat: 
`julia> length(unique([stat("I:\\").inode for i in 1:99]))`
`74` 
5. After a while the inode values will stay the same (at least on my machine): 
`julia> length(unique([stat("I:\\").inode for i in 1:99]))`
`1` 
```
julia> versioninfo()
Julia Version 1.9.2
Commit e4ee485e90 (2023-07-05 09:39 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 2 × AMD Ryzen 7 4700U with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver2)
  Threads: 1 on 2 virtual cores
```

**Comment by vtjnash** [comment]
The device number should be different, and the pair of those should form a unique identifier of the file. Is that not true in what you are seeing for Filesystem.samefile?

**Comment by sunito** [comment]
I've just checked ist: The device number is always the same: 0x0000000000010302.
`julia> length(unique([stat("I:\\").device for i in 1:99]))`
`1`

**Comment by sunito** [comment]
On Windows 7, the results are quite similar. 
In particular, the device number is the same (0x0000000000010302 = 66306)

**Comment by sunito** [comment]
I came across this bug when using `cp()` and `mv()` with  `force=true`

I'll include part of the original error message for others to find this issue:
**'src' and 'dst' refer to the same file/dir. This is not supported.**

**Comment by vtjnash** [comment]
libuv sets this value to the VolumeSerialNumber. Could you print those out and see if they are the same?https://github.com/libuv/libuv/blob/d09441ca03e399fe641da88624c8ea0476967187/src/win/fs.c#L1869
https://www.lifewire.com/how-to-find-a-drives-volume-label-or-serial-number-2624494

Possibly instead it should parse the Volume GUID. Could you confirm if this value looks unique:
https://www.wintips.org/how-to-find-volume-guid-list-volume-guids-per-disk/

**Comment by sunito** [comment]
Yes, the volume serial number is indeed the same as the stat.device number:   "0001-0302"

The method for retrieving the volume GUID only works for "normal" drives, not network drives.
(And, for my "C:"-partition, the value was different.)

But why do you ask about the device ids? The problem in my case is that the inode values are not stable.

**Comment by vtjnash** [comment]
That is strange. This is seems to be documented to be a guarantee by the NT kernel here: https://learn.microsoft.com/en-us/windows/win32/api/winbase/ns-winbase-file_id_info. But maybe this FILE_ID_128 is not the same as the LARGE_INTEGER one that is exposed by FileAllInformation or FILE_STAT_INFORMATION

**Comment by sunito** [comment]
Or maybe files on network drives are not regarded as regular files and theses structures are not supposed to have meaningful content. Just guessing.
